app/models/orchestration_template/orchestration_parameter.rb

```diff
@@ -26,9 +26,11 @@ class OrchestrationParameter
     attr_accessor :data_type
     attr_accessor :default_value
     attr_accessor :hidden
+    attr_accessor :required
     attr_writer   :constraints
 
     def initialize(hash = {})
+      @required = true
```

Comment:
@bzwei I feel like the default `required` value should be `false`.  This would also match the default when creating fields in the UI.  This would cause uses to explicitly state that they want to enable validation on the field.

Also need a test to validate the default value does not get changed.